public/app/features/plugins/extensions/getPluginExtensions.test.tsx

```diff
@@ -564,3 +575,178 @@ describe('getPluginExtensions()', () => {
+      expect(value.extensions).toHaveLength(2);
+      expect(value.extensions[0].pluginId).toBe(pluginId);
+      expect(value.extensions[1].pluginId).toBe(pluginId);
+    });
+  });
+
+  it('should emit the new state when the registries change', (done) => {
+    const observable = getObservablePluginExtensions({ extensionPointId });
+    const subscriber = jest.fn();
+
+    observable.subscribe(subscriber);
+
+    // Initial state
+    setTimeout(() => {
+      expect(subscriber).toHaveBeenCalledTimes(2);
+      expect(subscriber.mock.calls[1][0].extensions).toHaveLength(2);
+      expect(subscriber.mock.calls[1][0].extensions[0].pluginId).toBe(pluginId);
+      expect(subscriber.mock.calls[1][0].extensions[1].pluginId).toBe(pluginId);
+
+      // Register new link extension
+      pluginExtensionRegistries.addedLinksRegistry.register({
+        pluginId,
+        configs: [
+          {
+            title: 'Link 2',
+            description: 'Link 2 description',
+            path: `/a/${pluginId}/declare-incident`,
+            targets: extensionPointId,
+            configure: jest.fn().mockReturnValue({}),
+          },
+        ],
+      });
+
+      expect(subscriber).toHaveBeenCalledTimes(3);
+      expect(subscriber.mock.calls[2][0].extensions).toHaveLength(3);
+      expect(subscriber.mock.calls[2][0].extensions[0].pluginId).toBe(pluginId);
+      expect(subscriber.mock.calls[2][0].extensions[1].pluginId).toBe(pluginId);
+      expect(subscriber.mock.calls[2][0].extensions[2].pluginId).toBe(pluginId);
+
+      done();
+    }, 0);
+  });
+});
+
+describe('getObservablePluginLinks()', () => {
+  const extensionPointId = 'grafana/dashboard/panel/menu/v1';
+  const pluginId = 'grafana-basic-app';
+
+  beforeEach(() => {
+    pluginExtensionRegistries.addedLinksRegistry = new AddedLinksRegistry();
+    pluginExtensionRegistries.addedComponentsRegistry = new AddedComponentsRegistry();
+    pluginExtensionRegistries.addedLinksRegistry.register({
+      pluginId,
+      configs: [
+        {
+          title: 'Link 1',
+          description: 'Link 1 description',
+          path: `/a/${pluginId}/declare-incident`,
+          targets: extensionPointId,
+          configure: jest.fn().mockReturnValue({}),
+        },
+      ],
+    });
+
+    pluginExtensionRegistries.addedComponentsRegistry.register({
+      pluginId,
+      configs: [
+        {
+          title: 'Component 1',
+          description: 'Component 1 description',
+          targets: extensionPointId,
+          component: () => {
+            return <div>Hello world!</div>;
+          },
+        },
+      ],
+    });
+  });
+
+  it('should only emit the links', () => {
+    const observable = getObservablePluginLinks({ extensionPointId });
+
+    observable.subscribe((links) => {
+      expect(links).toHaveLength(1);
+      expect(links[0].pluginId).toBe(pluginId);
+      expect(links[0].type).toBe(PluginExtensionTypes.link);
+    });
+  });
```

Comment:
You might want to replace this with the `toEmitValuesWith` matcher as it has better handling of RxJs edge cases
```suggestion
  it('should only emit the links', async () => {
    const observable = getObservablePluginLinks({ extensionPointId }).pipe(first());

    await expect(observable).toEmitValuesWith((received) => {
      const links = received[0];
      expect(links).toHaveLength(1);
      expect(links[0].pluginId).toBe(pluginId);
      expect(links[0].type).toBe(PluginExtensionTypes.link);
    });
  });
```